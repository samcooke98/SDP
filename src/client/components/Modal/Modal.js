import React from "react";
import styles from "./web.css";

import Close from "react-icons/lib/md/close"

export default class Modal extends React.Component {
    componentWillMount() { 
        console.log("add event listener");
        try { 
            document.addEventListener("keydown", this.handleEscape);
        } catch(err) { 
            //Do nothing - It errors on the Server, as document isn't defined
        }
    }

    componentWillUnmount() { 
        document && document.removeEventListener("keydown", this.handleEscape);
    }

    render() {
        return (
            <div className={styles.modal} onKeyPress={this.handleEscape}>
                <div className={styles.container}>
                    <div className={styles.title}>
                        <h2>{this.props.label}</h2>
                        <Close height={32} width={32} onClick={this.props.onClose} />
                    </div><div className={styles.content}>
                        {this.props.children}
                    </div>
                </div>
            </div>
        )
    }

    handleEscape = (evt) => { 
        if(evt.keyCode === 27) { 
            this.props.onClose();
        }
    }


}


//In addition to CSS modules, you may wish to use: https://github.com/gajus/react-css-modules