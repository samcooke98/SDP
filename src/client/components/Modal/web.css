.container {
    width: 480px;
    display:flex;
    flex-direction: column;

    align-self: center;
}

.title {
    background-color: #2F80ED;
    max-height:102px;
    color: #F8F8F8;
    display: flex;
    flex-direction: row;
    padding: 24px;
    justify-content: space-between;
    align-items: center;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;   
    box-sizing:border-box;
}

.title h2 {
    display: inline;
    font-size:48px;
    font-family: "Raleway"

}

.content {
    background-color: #E6E9ED;
    min-height: 300px;
    padding: 24px;
}