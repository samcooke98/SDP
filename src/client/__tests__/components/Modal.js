import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import Modal from "../../components/Modal/Modal";


describe('React Component: Modal', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <Modal />   
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})