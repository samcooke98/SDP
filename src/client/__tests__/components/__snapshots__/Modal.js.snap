// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: Modal matches a snapshot 1`] = `
<div
  className={undefined}
  onKeyPress={[Function]}
>
   
  <div
    className="undefined undefined undefined"
  >
    <div
      className={undefined}
    >
      <h2 />
      <svg
        fill="currentColor"
        height={32}
        onClick={undefined}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "cursor": "pointer",
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width={32}
      >
        <g>
          <path
            d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
          />
        </g>
      </svg>
    </div>
    <div
      className={undefined}
    />
  </div>
</div>
`;
